internal/pkg/db/db.go

```diff
@@ -93,14 +101,81 @@ func (backend *Backend) AddDocument(document *sbom.Document) error {
 func (backend *Backend) GetDocumentByID(id string) (*sbom.Document, error) {
 	document, err := backend.Retrieve(id, nil)
 	if err != nil {
-		backend.Logger.Warn("Document could not be retrieved", "id", id, "err", err)
+		backend.Logger.Debug("Document could not be retrieved", "id", id, "err", err)
 
 		return nil, fmt.Errorf("failed to retrieve document: %w", err)
 	}
 
 	return document, nil
 }
 
+func (backend *Backend) GetDocumentByIDOrAlias(id string) (*sbom.Document, error) {
+	document, idErr := backend.GetDocumentByID(id)
+
+	if document == nil {
+		documents, aliasErr := backend.GetDocumentsByAnnotation(BomctlAnnotationAlias, id)
+		if aliasErr != nil || len(documents) == 0 {
```

Comment:
Should `len(documents) == 0` really be a failure case? Trying to mirror SQL behavior

Reply:
I was trying to be consistent with `backend.GetDocumentByID` which is being used in this same function, but I can change it.